File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
    this.findAll=function(){
        return $http.get('../brand/findAll.do');
    };

    this.add=function(entity){
        return $http.post("../brand/save.do",entity);
    }

    this.update=function(entity){
        return $http.post("../brand/update.do",entity);
    }

    this.findOne=function(id){
        return $http.get("../brand/findOne.do?id="+id);
    }

    this.dele=function(delIds){
        return $http.get("../brand/delete.do?ids="+delIds);
    }

    this.search=function(page,size,brand){
        return $http.post("../brand/search.do?page="+page+"&size="+size,brand)
    }
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 品牌服务层接口
 */
public interface BrandService {
    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    public void save(TbBrand brand);

    public TbBrand findOne(Long id);

    public void update(TbBrand brand);

    public void delele(Long[] ids);

    public PageResult findPage(TbBrand brand, int pageNum, int pageSize);

    public List<Map> selectBrandList();
}
